src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -306,26 +325,76 @@ export class StudyViewPageStore {
 
     readonly mutationProfiles = remoteData({
         await: ()=>[this.molecularProfiles],
-        invoke:()=>Promise.resolve(
-            this.molecularProfiles.result!.filter(profile => profile.molecularAlterationType === "MUTATION_EXTENDED")
-        ),
-        default: []
+        invoke: async ()=>{
+            return this.molecularProfiles.result.filter(profile => profile.molecularAlterationType === "MUTATION_EXTENDED")
+        },
+        default: [],
+        onResult:(mutationProfiles)=>{
+            if (!_.isEmpty(mutationProfiles)) {
+                this._chartVisibility.set(UniqueKey.MUTATED_GENES_TABLE, true);
+            }
+        }
     });
 
-    @computed
-    get cnaProfileIds() {
-        return this.molecularProfiles
+
+    readonly cnaProfiles = remoteData({
+        await: ()=>[this.molecularProfiles],
+        invoke: async ()=>{
+            return  this.molecularProfiles
             .result
             .filter(profile => profile.molecularAlterationType === "COPY_NUMBER_ALTERATION" && profile.datatype === "DISCRETE")
-            .map(profile => profile.molecularProfileId);
-    }
+        
+        },
+        default: [],
+        onResult:(cnaProfiles)=>{
+            if (!_.isEmpty(cnaProfiles)) {
+                this._chartVisibility.set(UniqueKey.CNA_GENES_TABLE, true);
+            }
+        }
+    });
 
     readonly clinicalAttributes = remoteData({
         await: () => [this.studies],
         invoke: () => defaultClient.fetchClinicalAttributesUsingPOST({
             studyIds: this.studies.result.map(study => study.studyId)
         }),
-        default: []
+        default: [],
+        onResult:(clinicalAttributes)=>{
+            // TODO: this is a temorary solution util we figure out the chartVisibility logic
+            let osStatusFlag = false;
+            let osMonthsFlag = false;
+            let dfsStatusFlag = false;
+            let dfsMonthsFlag = false;
+            let mutationCountFlag = false;
+            let fractionGenomeAlteredFlag = false;
+
+            clinicalAttributes.forEach(obj => {
+                if (obj.clinicalAttributeId === 'OS_STATUS') {
```

Comment:
Let's make all these strings to const, similarly what we have for MUTATION_COUNT